katello/hooks/pre/30-upgrade.rb

```diff
@@ -87,8 +84,8 @@ def step_path(step)
     upgrade_step :start_postgresql, :run_always => true
   end
 
-  if katello || foreman_proxy_content
-    upgrade_step :migrate_pulp, :run_always => true
+  if File.exist?(PULP2_MIGRATION_MARKER_FILE)
+    File.unlink(PULP2_MIGRATION_MARKER_FILE)
```

Comment:
Why unlink?

Reply:
https://puppet.com/docs/puppet/latest/type.html#exec-attribute-creates (currently direct links are broken) states:
> creates     => # A file to look for before running the command...

The exec runs `pulp-manage-db && touch /var/lib/pulp/init.flag`. That makes it idempotent. By removing the file, we trigger the exec and thus a database migration.